### PR TITLE
Merge psych master: Removed the bundled libyaml

### PR DESCRIPTION
**Pull request by hsbt**

### CODE REVIEW

**Comment by hsbt** [comment]
https://github.com/ruby/ruby-ci-image/pull/7

File: ext/psych/depend [hunk 1]
```diff
@@ -2,30 +2,10 @@ $(OBJS): $(YAML_H)
...
-emitter.o: yaml/config.h
-emitter.o: yaml/emitter.c
-emitter.o: yaml/yaml.h
-emitter.o: yaml/yaml_private.h
 loader.o: $(RUBY_EXTCONF_H)
-loader.o: yaml/config.h
-loader.o: yaml/loader.c
-loader.o: yaml/yaml.h
-loader.o: yaml/yaml_private.h
 parser.o: $(RUBY_EXTCONF_H)
```
Comment (nobu) [inline ext/psych/depend:8]: Also these lines are no longer used.

File: ext/psych/depend [hunk 2]
```diff
@@ -915,20 +891,7 @@ psych_yaml_tree.o: psych_parser.h
...
-reader.o: yaml/config.h
-reader.o: yaml/reader.c
-reader.o: yaml/yaml.h
-reader.o: yaml/yaml_private.h
 scanner.o: $(RUBY_EXTCONF_H)
-scanner.o: yaml/config.h
-scanner.o: yaml/scanner.c
-scanner.o: yaml/yaml.h
-scanner.o: yaml/yaml_private.h
 writer.o: $(RUBY_EXTCONF_H)
```
Comment (nobu) [inline ext/psych/depend:896]: Ditto.

**Comment by hsbt** [comment]
@shyouhei I'm not sure why crossbuild was failed with the missing `yaml/yaml.h` error. 

https://github.com/ruby/ruby/runs/5667870754?check_suite_focus=true#step:12:544

Can you have a point of this failure?

**Comment by shyouhei** [comment]
Seems like a CI environment issue.  Let me take a look...

**Comment by hsbt** [comment]
Thanks. I confirmed arm64 package of `libyaml*` with https://github.com/ruby/ruby-ci-image/pull/7

* https://github.com/ruby/ruby-ci-image/runs/5659559982?check_suite_focus=true#step:6:2210
* https://github.com/ruby/ruby-ci-image/runs/5659559982?check_suite_focus=true#step:6:2410

**Comment by shyouhei** [comment]
@hsbt it seems the package has `/usr/include/yaml.h` but not `yaml/yaml.h`.
https://packages.ubuntu.com/focal/arm64/libyaml-dev/filelist

https://github.com/ruby/ruby/blob/4e6b49099c5fda02e800e6daaf5cc4de5f53b90a/ext/psych/extconf.rb#L10

In the above line, 
- `enable_config("bundled-libyaml")` is not set (false)
- `find_header('yaml.h')` is OK,
- `find_library('yaml', 'yaml_get_version')` is failing.

So in short we need not only libyaml-dev but also ARM64 version of libyaml.so somewhere inside of the container.

ARM is kind of okay because Ubuntu has ARM version (we can add apt line), but what about mingw...?


Let me think for a while.

**Comment by shyouhei** [comment]
I propose disabling those corss-compilers in a meantime.  Cirrus CI is checking ARM anyways so it can be acceptable.

**Comment by hsbt** [comment]
I got the same errors before removing `yaml/yaml.h` dependency on depend file at [c40a06c](https://github.com/ruby/ruby/pull/5693/commits/c40a06c2d0adc999406b8aafda40fc1d19cb550d). 

Therefore, I'm not sure why it still happened. 

I will disable them with your proposal. Thanks.

**Review by shyouhei — APPROVED** [review]

**Comment by deivid-rodriguez** [comment]
Not sure if it helps or if it's actually correct but I was getting the same error when building on MacOS and I fixed it with

```diff
diff --git a/ext/psych/extconf.rb b/ext/psych/extconf.rb
index 857f8e68c4..7da73760e3 100644
--- a/ext/psych/extconf.rb
+++ b/ext/psych/extconf.rb
@@ -7,7 +7,7 @@
 
 dir_config 'libyaml'
 
-if enable_config("bundled-libyaml", false) || !(find_header('yaml.h') && find_library('yaml', 'yaml_get_version'))
+if enable_config("bundled-libyaml", false) || !pkg_config('yaml-0.1') && !(find_header('yaml.h') && find_library('yaml', 'yaml_get_version'))
   # Embed libyaml since we could not find it.
 
   $VPATH << "$(srcdir)/yaml"
```

**Comment by peterzhu2118** [comment]
I'm also running into this issue on macOS and @deivid-rodriguez's patch fixes it for me.
